File: February 2026 GFG SOLUTION/February-07.java
class Solution {
    int maxSum(int[] arr) {
        int n = arr.length, s = 0, cur = 0, mx;
        for (int i = 0; i < n; i++) { s += arr[i]; cur += i * arr[i]; }
        mx = cur;
        for (int i = 0; i < n - 1; i++) {
            cur += n * arr[i] - s;
            mx = Math.max(mx, cur);
        }
        return mx;
    }
}

File: February 2026 GFG SOLUTION/February-07.py
class Solution:
    def maxSum(self, arr):
        n = len(arr)
        s = sum(arr)
        cur = sum(i * arr[i] for i in range(n))
        mx = cur
        for i in range(n - 1):
            cur += n * arr[i] - s
            mx = max(mx, cur)
        return mx
